### PR TITLE
Creating dynamo table for install activity data

### PR DESCRIPTION
**Pull request by manasaV3**
Addresses https://github.com/chanzuckerberg/napari-hub/issues/867

## Description

The PR adds infrastructure changes to create a dynamoDb table for install activity with the required `hash_key=plugin_name` and `range_key=type_timestamp`.  It also provides the backend API with permission to query the table.

### CODE REVIEW

**Comment by potating-potato** [comment]
shouldn't these tables be managed by internal infra and one per environment?

**Comment by potating-potato** [comment]
> @potating-potato hmm so would this be better suited in https://github.com/chanzuckerberg/sci-imaging-infra?

Yeah that's what I am wondering about. I don't have a concrete suggestion here either but would ask Manasa to think about the tradeoff and make a decision

**Comment by manasaV3** [comment]
> shouldn't these tables be managed by internal infra and one per environment?
@potating-potato  As the dynamo stream from some of the future tables will trigger the lambda, having the tables one per environment is no longer a viable solution.  :(
Instead, we will maintain a single dev environment that the team will share.

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 1]
```diff
@@ -97,6 +97,26 @@ module frontend_service {
...
+  attributes          = [
+                          {
+                            name = "plugin_name"
+                            type = "S"
+                          },
+                          {
+                            name = "type_timestamp"
+                            type = "S"
+                          }
+                        ]
```
[the remark this answers is not in the document]
Comment (manasaV3) [reply .happy/terraform/modules/ecs-stack/main.tf:114]: We will only define the primary key combination as it is a requirement of dynamo to configure the keys at the time of table creation. All of the other attributes can be added at runtime.

[the remark this answers is not in the document]
Comment (potating-potato) [reply .happy/terraform/modules/ecs-stack/main.tf:114]: outside of the scope for this PR, if you intend to manage the schema outside of the terraform, you would have to think about schema migrations and toolings to keep it consistent, though a lot lighter than a SQL database, it is still possible to have conflict in schemas that needs to be resolved. Not sure how your team is planning to track and address these, but raising here for now

**Review by potating-potato — APPROVED** [review]

File: .happy/terraform/modules/ecs-stack/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (manasaV3) [reply .happy/terraform/modules/ecs-stack/main.tf:114]: We were planning to leverage the data modeling ORMs to ensure the schemas are consistently used. :)

**Pull request merged**